### PR TITLE
Signing

### PR DESCRIPTION
**Pull request by mtrmac**
Note: This depends on #24; ATM #24 is included via a merge in this PR, I will rebase this branch after #24 is merged.

### CODE REVIEW

**Comment by runcom** [comment]
#24 merged Miroslav, but CI has issues here

**Comment by mtrmac** [comment]
Fixed, ready for merging.

File: Makefile [hunk 1]
```diff
@@ -53,8 +53,9 @@ shell: build-container
 
 check: validate test-unit test-integration
 
+# The tests can run out of entropy and block in containers, so replace /dev/random.
 test-integration: build-container
-	$(DOCKER_RUN_DOCKER) hack/make.sh test-integration
+	$(DOCKER_RUN_DOCKER) bash -c 'rm -f /dev/random; ln -sf /dev/urandom /dev/random; hack/make.sh test-integration'
```
Comment (runcom) [inline Makefile]: what's this one?

Comment (mtrmac) [inline Makefile]: See the comment just above :)

There is apparently no way (or no reasonable way?) to convince GPG not to use `/dev/random`, and in the Travis environment we don't have write access to the RNG so we can’t even send fake entropy in there.

But the container is all ours to play with, so we can point `/dev/random` to the non-blocking `urandom` device, which fixes the issue and does not affect the surrounding environment.

Does this need a better comment?

Comment (runcom) [inline Makefile]: nah, it's fine I just overlooked the comment

**Comment by runcom** [comment]
@mtrmac I'm going ahead and merge, good to you again?

File: Makefile (hunk 1 above)
Comment (mtrmac) [inline Makefile]: (This is the fix to the CI failures BTW.)

**Comment by mtrmac** [comment]
@runcom Yes please!

**Pull request merged**